server/mainapp/models.py

```diff
@@ -123,3 +123,183 @@ def fetch_data(self) -> response.JsonResponse:
             return docs
 
         raise DataFetchingError("There Are No Posts In The Database At This Moment")
+
+
+class Likes:
+    def __init__(self) -> None:
+        """
+        Connect to MongoDB
+        """
+        client = pymongo.MongoClient(DATABASE["mongo_uri"])
+        self.db = client[DATABASE["db"]]["Likes"]
+
+    def insert_data(
+            self,
+            music_data_pid: str,
+            user_email: str,
+    ) -> None:
+        """Insert file name and data into db
+
+        Args:
+            music_data_pid: FK for pid of the post
+            user_email: FK for user email
+
+        Returns:
+            None
+        """
+        if self.db.find_one({"music_data_pid": music_data_pid, "user_email": user_email}):
+            raise FileAlreadyExistsForCurrentUserError(
+                f"Like for post {music_data_pid} Already Exists For User {user_email}"
+            )
+
+        data = {
+            "music_data_pid": music_data_pid,
+            "user_email": user_email,
+        }
+        self.db.insert_one(data)
+
+    def delete_data(self, music_data_pid: str, user_email: str) -> None:
+        """Delete file and related data from db
+
+        Args:
+            music_data_pid: User Email ID
+            user_email: Name of File
+
+        Returns:
+            None
+        """
+        if self.db.find_one(
+                {
+                    "music_data_pid": music_data_pid,
+                    "user_email": user_email,
+                }
+        ):
+            self.db.delete_one(
+                {
+                    "music_data_pid": music_data_pid,
+                    "user_email": user_email,
+                },
+            )
+        else:
+            raise FileDoesNotExistForCurrentUserError(
+                f"Like for post {music_data_pid} does not exists for user {user_email}"
+            )
+
+    def delete_all_data(self) -> None:
+        """Delete all likes from db
+
+                Args:
+                    None
+
+                Returns:
+                    None
+                """
+        self.db.remove()
+
+    def fetch_data(self) -> response.JsonResponse:
+        """Fetches all data from db
+
+        Args:
+            None
+
+        Returns:
+            response.JsonResponse
+        """
+        if data := self.db.find(
+                {},
+                {
+                    "_id": 0,
+                },
+        ).sort("music_data_pid", -1):
+            likes = list(data)
+            return likes
+
+        raise DataFetchingError("There Are No Likes In The Database At This Moment")
+
+
+class Comments:
```

Comment:
Similar to the likes - To receive the comments, a URL will need to be created along with a view. This will refer to the endpoint that accepts the comment and performs the database functionality - It will also update the UI with the comment for the client